edward/inferences/inference.py

```diff
@@ -188,7 +188,11 @@ def run(self, logdir=None, variables=None, use_coordinator=True,
     self.initialize(*args, **kwargs)
 
     if logdir is not None:
+      self.logging = True
       self.train_writer = tf.train.SummaryWriter(logdir, tf.get_default_graph())
+      self.summarize = tf.merge_all_summaries()
+    else:
+      self.logging = False
```

Comment:
the boolean flag makes sense. we also need to somehow enable this option when the user manually controls inference and does not call `inference.run()`. see, e.g., [`normal_idiomatic_tf.py`](https://github.com/blei-lab/edward/blob/master/examples/normal_idiomatic_tf.py), [`vae.py`](https://github.com/blei-lab/edward/blob/master/examples/vae.py).